lib/compiler/test/bs_match_SUITE.erl

```diff
@@ -3220,6 +3222,29 @@ gh_6923(_Config) ->
 do_gh_6923([<<"abc">>, A]) when is_integer(A) -> first;
 do_gh_6923([<<"abc">>, A]) when is_tuple(A) -> second.
 
+bs_test_tail(Config) ->
+    Bin = term_to_binary(Config),
+
+    ok = bs_test_tail_skip(Bin),
+
+    "abc" = bs_test_tail_int(<<(id(<<"abc">>))/binary>>),
+
+    [2.0,3.0] = bs_test_tail_float(<<(id(2.0)):64/float,(id(3.0)):64/float>>),
+    {'EXIT',{function_clause,_}} = catch bs_test_tail_float(<<(id(-1)):128>>),
+
+    ok.
+
+bs_test_tail_skip(<<_, T/binary>>) -> bs_test_tail_skip(T);
+bs_test_tail_skip(<<>>) -> ok.
+
+bs_test_tail_int(<<H:8, T/binary>>) ->
+    [H|bs_test_tail_int(T)];
+bs_test_tail_int(<<>>) -> [].
+
+bs_test_tail_float(<<F:64/float, T/binary>>) ->
+    [F|bs_test_tail_float(T)];
+bs_test_tail_float(<<>>) -> [].
```

Comment:
I had to do a manual test to convince myself it would work correctly in such a case (and the optimisation wouldn't be applied), perhaps it's worth adding something like this to the test suite?

```erlang
bs_test_partial_tail(<<0:8, T/binary>>) -> bs_test_partial_tail(T);
bs_test_partial_tail(<<>>) -> ok.
```

Reply:
Good idea. Added.